test/membership-checksum.js

```diff
@@ -31,9 +31,47 @@ function generateChecksumString(members) {
         })
         .map(function (member) {
             var address = member.address || (member.host + ':' + member.port);
-            return address +
-                    member.status +
-                    member.incarnationNumber
+
+            var labelChecksum = 0;
+
+            if (member.labels) {
+                Object.keys(member.labels).forEach(function (key) {
+                    var keyBuffer = new Buffer(key);
```

Comment:
buffers in nodejs can't be resized after initialization. Since the size of the required buffer is unknown due to the encoding of strings this was the easiest way of calculating the final buffer.